### PR TITLE
New api

### LINKED ISSUE #6: new api
Figured this might be easier to read / figure out what's going on:
```js
var multipart = require('multipart-read-stream')
var http = require('http')
var pump = require('pump')

http.createServer(function (req, res) {
  var type = req.headers['content-type']
  var filestream = multipart(type, fileHandler)

  pump(req, filestream, function (err) {
    if (err) (res.statusCode = 400 && res.end('oh no!'))
    res.end()
  })

  function fileHandler (fieldname, file, filename) {
    console.log(`reading file ${filename} from field ${fieldname}`)
    file.pipe(process.stdout)
  }
}).listen(8080)
```
cc/ @davidmarkclements thoughts?

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Implements the new api. Closes #6. Enables https://github.com/yoshuawuyts/merry/issues/20 Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,25 +6,27 @@ Read a multipart stream over HTTP. Built on top of [pez][pez].
...
     res.end()
   })
-  req.pipe(rs)
 
-  function fileHandler (fieldname, file, filename) {
-    console.log(`reading file ${filename} from field ${fieldname}`)
-    file.pipe(process.stdout)
+  function handler (fieldname, file, filename) {
+    console.log('reading file ' + filename + ' from field ' + fieldname)
+    file.pipe(fs.createWriteStream(path.join('/tmp', filename)))
```
Comment (davidmarkclements) [inline README.md]: any reason in example we use both `pump` and `pipe`?

File: README.md [hunk 2]
```diff
@@ -6,25 +6,27 @@ Read a multipart stream over HTTP. Built on top of [pez][pez].
...
   }
-})
+}).listen(8080)
 ```
 
 ## API
-### readableStream = multipart(req, res, fileHandler, options?, done)
+### readableStream = multipart(, fileHandler, options?, done)
```
Comment (davidmarkclements) [inline README.md]: got a leading comma, assume that should say "headers" or such
Also we've removed the `done` cb

**Review by davidmarkclements — CHANGES_REQUESTED** [review]
great - love it

File: README.md (hunk 1 above)
Comment (yoshuawuyts) [reply README.md]: oops

**Comment by yoshuawuyts** [comment]
Updated :sparkles:

**Pull request merged**